### PR TITLE
Pipe read

### PR DESCRIPTION
**Pull request by breakreturn**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,6 +30,10 @@ The following cmake options are available:
 
 * `ENABLE_LOG_TO_FILE` - makes `dbus-proxy` log to "/tmp/dbus-proxy.log"
 
+Please note that the `ENABLE_LOG_TO_FILE` option should not be used in other
+contexts that trouble shooting and debugging. It's not suitable for production
```
Comment (tobsan) [inline README.md]: troubleshooting is one word

File: README.md [hunk 2]
```diff
@@ -54,6 +58,12 @@ You can then interact with the socket via, for instance D-Feet or dbus-send.
 
 Configuration files
 -------------------
+Please note that the below description is for when running `dbus-proxy` manually,
+when it might be convenient to redirect a file with the config. Generally `dbus-proxy`
+is meant to be spawned bt another process that writes to stdin of `dbus-proxy`. When
```
Comment (tobsan) [inline README.md]: spawned by*

File: src/proxy.c [hunk 3]
```diff
@@ -23,14 +23,24 @@
 
 
 #include "proxy.h"
-#include <signal.h>
+
 #include <stdio.h>
 #include <stdlib.h>
+#include <stdlib.h>
```
Comment (tobsan) [inline src/proxy.c]: double include

**Review by rpannek — APPROVED** [review]

File: src/proxy.c [hunk 4]
```diff
@@ -787,16 +856,27 @@ int main(int argc, char *argv[]) {
...
+    g_io_add_watch(channel,
+                   G_IO_IN | G_IO_PRI | G_IO_ERR | G_IO_HUP,
+                   (GIOFunc)stdin_watch,
+                   section);
+
+    g_message("Entering mainloop\n");
+
+    /* Start listening */
+    start_bus();
+    mainloop = g_main_loop_new(NULL, FALSE);
```
Comment (tobsan) [inline src/proxy.c]: These magic values could deserve some comment.

**Review by tobsan — CHANGES_REQUESTED** [review]

File: src/proxy.c [hunk 5]
```diff
@@ -703,7 +700,10 @@ gboolean log_file_is_open() {
 }
 
 gboolean open_log_file() {
-    log_file = fopen("/tmp/dbus-proxy.log", "a");
+    char buf[100] = {0};
```
Comment (tobsan) [inline src/proxy.c]: This feels unnecessarily large.

**Review by tobsan — APPROVED** [review]
Just a comment, looks good to me.

**Review by frznlogic — APPROVED** [review]

File: src/proxy.c [hunk 6]
```diff
@@ -628,70 +644,51 @@ void start_bus() {
...
-        /* We have atleast one root object in file */
-        config_found = 1;
+void parse_full_config(const char *config_string, const char *section) {
+    json_error_t error;
+    json_t *root;
+    json_t *config;
 
-        // Get array
-        config = json_object_get (root, full_section);
+    char *full_section = calloc(sizeof(char), 30);
```
Comment (kursatkobya) [inline src/proxy.c]: better to use 
`calloc(30, sizeof(char));`
since the first argument indicates number of elements

File: src/proxy.c [hunk 7]
```diff
@@ -769,14 +847,30 @@ int main(int argc, char *argv[]) {
...
+#ifdef LOG_TO_STDOUT
+    silence_logging = FALSE;
+#endif
+    if (silence_logging) {
+        g_log_set_handler(NULL /*use default log domain*/,
+                          G_LOG_LEVEL_MASK,
+                          log_handler_silent,
+                          NULL /*no need to pass data to handler*/);
+    }
+
```
Comment (kursatkobya) [inline src/proxy.c]: maybe getting rid of a new variable would be nicer. So instead `#ifdef LOG_TO_STDOUT` you can add 
```
#ifndef LOG_TO_STDOUT
    g_log_set_handler(NULL, G_LOG_LEVEL_MASK, log_handler_silent, NULL);
#endif
```

File: CMakeLists.txt [hunk 8]
```diff
@@ -37,12 +37,17 @@ set(VERSION ${${PROJECT_NAME}_MAJOR_VERSION}.${${PROJECT_NAME}_MINOR_VERSION}.${
...
+option(ENABLE_LOG_TO_FILE "Enables logging to file" OFF)
+option(ENABLE_LOG_TO_STDOUT "Enables logging to stdout/stderr" OFF)
 
 if(ENABLE_LOG_TO_FILE)
     add_definitions(-DLOG_TO_FILE)
 endif()
 
+if(ENABLE_LOG_TO_STDOUT)
+    add_definitions(-DLOG_TO_STDOUT)
+endif()
```
Comment (kursatkobya) [inline CMakeLists.txt]: so is it possible to undeclare LOG_TO_FILE in this case

**Review by kursatkobya — APPROVED** [review]
Looks good, i have just minor comments.

**Pull request merged**